### PR TITLE
feat: add hidden skip-to-content button

### LINKED ISSUE #47523: Skip to content Button for accessibility
### Is your feature request related to a problem? Please describe.

I was trying to implement freeCodeCamp `Skip to content` button to Chapter, but there is none in freeCodeCamp. The button is used in websites like YouTube, this button will provide quality of life to people who access the site daily and give them chance to skip some annoys

### Describe the solution you'd like

Creating a skip to content button component, that skips the navigation bar

### Describe alternatives you've considered

Toggle the nav bar component, that hides the navigation. Although not standard, it's used in sites like CodePen

### PR DESCRIPTION
**Pull request by diivi**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #47523 

Working:
skip to content button in action

### CODE REVIEW

**Comment by diivi** [comment]
This will require an id of #content-start on the first heading/ result of the news page too (search results). I was asked to mention this here, for more info - https://discord.com/channels/692816967895220344/715074489422970962/1022137562271330314

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
HI @diivi thanks for putting this together!

As I mentioned in discord, there's no need to create multiple ids to jump to.  The default layout wraps every page, so you can jump to an id inside that.

If we go this route we basically guarantee an inconsistent experience. Future maintainers are likely to forget to add the `id='content-start'` to a new type of page.

**Review by ojeytonwilliams — DISMISSED** [review]
Thanks, @diivi, this works really well. LGTM :+1:

**Comment by bbsmooth** [comment]
> there's no need to create multiple ids to jump to

@ojeytonwilliams I'm going to disagree a little here. For the Steps, the skip link should focus on the `h1` in the instructions. If it focuses on the `#content-start` div that wraps `main` instead, then the user will still have to tab through the bread crumb links and editor/console/preview buttons. 

My suggestion is to add an `id` to the `h1` and have the skip link jump to that.

**Comment by ojeytonwilliams** [comment]
Thanks @bbsmooth, good catch! I agree that it should skip all the navigation, not some of it.  However, the root of the problem is that the navigation components are scattered across the site.

To keep things DRY and reasonably easy to understand, I think it would be better to pull the BreadCrumbs up to the `default-layout` rather than pushing the target ids down into 8 components.  This may require a bit of Gatsby magic, but it should be doable.

**Comment by diivi** [comment]
I don't think moving `ActionRow` from `DesktopLayout` to `DefaultLayout` will be that straightforward because of all the props that it requires, I even tried doing so. 
Could you point me in the right direction explaining how I can pull the `ActionRow` component out of `DesktopLayout` component?
desktoplayout.tsx

**Comment by ojeytonwilliams** [comment]
Sure. The Gatsby [docs](https://www.gatsbyjs.com/docs/programmatically-create-pages-from-data/#specifying-a-template) are pretty good, but the main thing you need to do is make the data available inside `layout-selector.tsx`.  That can be achieved via a GraphQL query which Gatsby will run and then inject the result as a `data` prop.

So, I'd use the approach in client/src/templates/Challenges/classic/show.tsx as a guide.

Finally, there's no need to move the entire ActionRow, just the BreadCrumb part.  Reason being, the goal is to put all the navigation elements in the DefaultLayout, but the BreadCrumb is the only part of ActionRow that deals with navigation.

I hope that helps!

**Comment by bbsmooth** [comment]
> there's no need to move the entire ActionRow

I personally think that the skip to content link should skip both the bread crumbs and the buttons in the ActionRow, but there is no hard rule on where it should skip to. If we do feel the buttons are part of the main content and shouldn't be skipped then I suggest we might change the text on the skip link to "Skip navigation" which I think would more accurately reflect what the link is doing here.

**Comment by diivi** [comment]
I elevated the main breadcrumb navigation, it's now a part of the default layout component.
fcc

**Comment by ojeytonwilliams** [comment]
@diivi this is looking great, really nice work.  The only issue I'm running into is that if I tab backwards after skipping past the breadcrumbs, the breadcrumbs receive focus as expected, but they don't scroll into view.  Other pages like /learn, don't have this problem - the page scrolls as I tab around.

Having said all that, this is already present on main, so we've not made things worse.  As such, I'm happy to merge this is if @bbsmooth is and we can deal with the lack of scrolling another time.

File: client/src/components/Header/index.tsx
Comment (Sboonny) [inline client/src/components/Header/index.tsx:91]: there is no need to DRY the button, it should be used only once and if it's used multiple times in the future, then we need to change the structure of the page instead of DRY the button.

```suggestion
<a href='#content-start' className='skip-to-content-button'>Skip To Content</a>
```

File: client/src/components/Header/index.tsx
Comment (Sboonny) [inline client/src/components/Header/index.tsx:9]:
```suggestion
```

File: client/src/components/Header/components/skip-to-content-button.tsx
Comment (Sboonny) [inline client/src/components/Header/components/skip-to-content-button.tsx:24]: You can delete this component and move the anchor to `index.tsx`, reason:

- using JavaScript for styling is dangerous because if JavaScript break the whole page break, while the site can function with broken CSS fine

- it's better for the performance to use CSS

```suggestion
```

File: client/src/components/Header/components/skip-to-content-button.css
Comment (Sboonny) [inline client/src/components/Header/components/skip-to-content-button.css:10]: the `index.tsx` page is using `header.css`, so there is no need to create new file

```suggestion
```

if you decide to move it, could you change the styles too?

You can add some basic styles for now, maybe I will discuss with ahmad how to style it, but that is for different PR.

```css
.skip-to-content-button {
  position: absolute;
  top: var(--header-height);
  border-radius: 15px;
  font-weight: 600;
  padding-block: 1em;
  padding-inline: 1.5em;
  left: -1000px;
}

.skip-to-content-button:focus {
  left: 0px;
}
```

**Review by Sboonny — CHANGES_REQUESTED** [review]
Thank you for the awesome work, here are some nitpicks and some refactoring

**Comment by ojeytonwilliams** [comment]
@ahmadabdolsaheb as discussed in Discord, these code changes result in duplicate breadcrumbs on all but the new project-based curriculum.  Like so:

image

and

image


Would it be okay if simply deleted the smaller ones? That way it's a consistent experience across all our challenge pages.

**Comment by diivi** [comment]
Waiting for @ahmadabdolsaheb to comment on this to continue.

**Comment by ahmadabdolsaheb** [comment]
Yes, the smaller breadcrumb can go.

**Comment by Sboonny** [comment]
@diivi I am wondering, is the skip navigation changes related to breadcrumb changes? 

Because it's hard to distinguished "why the test are failing", if it's possible to create two PRs, so reviewing and fixing can be easier that would be awesome 👍 . Saying that, maybe I am wrong and breadcrumb is necessary, after all I only skimmed through breadcrumbs discussion

here is what I found reading cypress errors

- the new `breadcrumbs` are stacking on top of `textarea`
- the new elements have changed how the old elements structured (html structure), if people are okay with the new hierarchy, running `npm run test -- -u` or `npx jest -u` should create a new snapshot  

I have no idea why localization test fails 🤷, I just hope fixing the other two will clear it as well.

**Comment by diivi** [comment]
I was asked to make the breadcrumbs part of the navigation and not the main content. So I moved them. But then I noticed, on some pages, there were two breadcrumbs, so I removed the second ones with this change. This makes skip to content skip the breadcrumbs too, which shouldn't be a part of the main content but only the navigation bar/header. This is why the two changes are part of the same PR, because if I just add skip to content, it will skip to the breadcrumbs. Should I revert to the point where I just added the skip to content button and skip the breadcrumb navigation in another PR?

**Comment by naomi-lgbt** [comment]
No, IMHO the removal of the old breadcrumbs is within the scope of this PR and should stay.

**Comment by Sboonny** [comment]
@diivi thank you for more context, it looks like related to skip to the button, so please keep it one PR. sorry for the confusion 👍

Please run one of the snapshot generating command, and fix the first issue however you see fit.

Although I have already seen you in discord and there is no need to link this again, but here it's anyway

If you need help, please join us in [our chat room](https://discord.gg/PRyKn3Vbay) or [the forum](https://forum.freecodecamp.org/c/contributors/3).

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1: 

Nice work, @diivi!

**Comment by bbsmooth** [comment]
This works really well. LGTM :+1:

**Review by Sboonny — APPROVED** [review]

**Comment by Sboonny** [comment]
Thank you for the awesome work

**Pull request merged**
